raiden_contracts/contracts/services/ServiceRegistry.sol

```diff
@@ -27,4 +29,20 @@ contract ServiceRegistry is Utils {
         // Transfer the deposit to the smart contract
         require(token.transferFrom(msg.sender, address(this), amount));
     }
+
+    /// Set the URL used to access a service via HTTP.
+    /// When this is called for the first time, the service's ethereum address
+    /// is also added to `service_addresses`.
+    function setURL(string memory new_url) public {
+        require(bytes(new_url).length != 0);
+        if (bytes(urls[msg.sender]).length == 0) {
+            service_addresses.push(msg.sender);
```

Comment:
Is this a constant-gas operation?

Reply:
Not an official source, but according to https://ethereum.stackexchange.com/questions/30843/is-the-cost-of-pushing-proportional-to-the-arrays-size and https://www.reddit.com/r/ethereum/comments/4usq1q/is_the_cost_of_arraypush_a_constant/d5sn2lb/ it is.